### ISSUE TITLE
Bot performance when using subcommands.

### ISSUE THREAD

**Issue by scriptdaemon** [yargs/yargs#788]
While writing a chat bot using subcommands, there's a very noticeable delay between issuing the bot a command and actually getting a response from the bot. I also notice in very naive tests (i.e. only running the bot methods and not how the bot behaves when issued a command) that the command object doesn't get covered, which led me to find that the builder gets called in `runCommand()`.

```js
this._parser = yargs()
this._parser.command({
  'command': '!ch <cmd>',
  'builder': yargs => {
    yargs.command(dice(this))
    yargs.command(uptime(this))
  }
})
```

I'm wondering if it would be possible to apply the builder only once when the command is added instead of every time it is run. @bcoe What do you think?

**Comment by bcoe** [comment]
@scriptdaemon I know some folks appreciate the fact that yargs' builder/handler are invoked lazily -- since it allows them to conditionally require dependencies when the methods are invoked ... maybe we could get the best of both worlds by introducing the concept of a cache? don't re-runt he builder if its been run before (but don't pre-run it).

Unfortunately, this might take a bit of refactoring; the command module is getting a little bit gnarly -- thoughts and help appreciated.

**Comment by shadowspawn** [comment]
This issue has not had any activity in years. It isn't likely to get acted on due to this report.

Feel free to open a new issue if it comes up again, with new information and renewed interest.
 
Thank you for your contributions.
